mainauto.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include "display.h"
+#include "mapio.h"
+#include "datastrc.h"
+#include "control.h"
+
+int main(int argc, char &argv){
+
+allocate_memory(*gmdt);
+//First allocate some memory
+
+allocate_pngns(*gmdt);
+//allocate memory for penguins
+
+read_data(argv[1], *gmdt);
```

Comment:
This is correct, but above this, check whether there are enough arguments. At this moment, if the user passes no arguments, `argv[1]` might not even exist (the array will only contain `argv[0]`). Just check if `argc` is bigger or equal to 3.